### ISSUE TITLE
System.IndexOutOfRangeException with AltCover 8.3.839

### ISSUE THREAD

**Issue by Yana-Yaa** [SteveGilham/altcover#167]
Hi, I am trying to make code coverage for one complicated Windows App by using .NET Framework 4.7.2 and NUnit and two other test frameworks. The purpose to use Altcover is to find out the coverage for C# code. When I executed altcover.exe to copy assemblies(the number is over 300), I got IndexOutOfRangeException. 
The AltCover version is 8.3.839
I found out one similar issue, [AltCover issue 135](https://20170801.azurewebsites.net/index.php?q=oKipp7eAc2SYqrfXwMue06bScLamuOXak9nS1q28m9Kjt7vJrM7dnrC6ueKbpWShfpo). But it doesn't help, so I decide to create a new issue for my case. 
```
Unhandled Exception: System.IndexOutOfRangeException: Index was outside the bounds of the array.
   at Mono.Cecil.Cil.CodeReader.ReadOpCode()
   at Mono.Cecil.Cil.CodeReader.ReadCode()
   at Mono.Cecil.Cil.CodeReader.ReadFatMethod()
   at Mono.Cecil.Cil.CodeReader.ReadMethodBody()
   at Mono.Cecil.Cil.CodeReader.ReadMethodBody(MethodDefinition method)
   at Mono.Cecil.MethodDefinition.<>c.<get_Body>b__41_0(MethodDefinition method, MetadataReader reader)
   at Mono.Cecil.ModuleDefinition.Read[TItem,TRet](TRet& variable, TItem item, Func`3 read)
   at Mono.Cecil.MethodDefinition.get_Body()
   at Mono.Cecil.MethodDefinition.get_DebugInformation()
   at AltCover.ProgramDatabase.getModuleDocuments@315-2.Invoke(MethodDefinition m) in /_//AltCover.Engine/ProgramDatabase.fs:line 315
   at Microsoft.FSharp.Collections.Internal.IEnumerator.map@75.DoMoveNext(b& curr) in D:\a\_work\1\s\src\fsharp\FSharp.Core\seq.fs:line 78
   at Microsoft.FSharp.Collections.Internal.IEnumerator.MapEnumerator`1.System.Collections.IEnumerator.MoveNext() in D:\a\_work\1\s\src\fsharp\FSharp.Core\seq.fs:line 64
   at Microsoft.FSharp.Core.CompilerServices.RuntimeHelpers.takeOuter@307[T,TResult](ConcatEnumerator`2 x, Unit unitVar0) in D:\a\_work\1\s\src\fsharp\FSharp.Core\seqcore.fs:line 308
   at Microsoft.FSharp.Collections.Internal.IEnumerator.map@75.DoMoveNext(b& curr) in D:\a\_work\1\s\src\fsharp\FSharp.Core\seq.fs:line 77
   at Microsoft.FSharp.Collections.Internal.IEnumerator.MapEnumerator`1.System.Collections.IEnumerator.MoveNext() in D:\a\_work\1\s\src\fsharp\FSharp.Core\seq.fs:line 64
   at Microsoft.FSharp.Collections.SeqModule.DistinctBy@1109.GenerateNext(IEnumerable`1& next) in D:\a\_work\1\s\src\fsharp\FSharp.Core\seq.fs:line 1110
   at Microsoft.FSharp.Core.CompilerServices.GeneratedSequenceBase`1.MoveNextImpl() in D:\a\_work\1\s\src\fsharp\FSharp.Core\seqcore.fs:line 405
   at Microsoft.FSharp.Collections.SeqModule.ToList[T](IEnumerable`1 source) in D:\a\_work\1\s\src\fsharp\FSharp.Core\seq.fs:line 803
   at AltCover.OpenCover.visitModule@140-13(FSharpFunc`2 recordFile, OpenCoverContext s, ModuleEntry moduleDef) in /_//AltCover.Engine/OpenCover.fs:line 162
   at AltCover.Visitor.stateful@1518-1.Invoke(T node) in /_//AltCover.Engine/Visitor.fs:line 1520
   at Microsoft.FSharp.Primitives.Basics.List.map[T,TResult](FSharpFunc`2 mapping, FSharpList`1 x) in D:\a\_work\1\s\src\fsharp\FSharp.Core\local.fs:line 246
   at Microsoft.FSharp.Collections.SeqModule.Fold[T,TState](FSharpFunc`2 folder, TState state, IEnumerable`1 source) in D:\a\_work\1\s\src\fsharp\FSharp.Core\seq.fs:line 731
   at AltCover.Visitor.visit(IEnumerable`1 visitors, IEnumerable`1 assemblies) in /_//AltCover.Engine/Visitor.fs:line 1506
   at AltCover.Main.I.result@722.Invoke(Unit unitVar0) in /_//AltCover.Engine/Main.fs:line 745
   at AltCover.PathOperation.DoPathOperation[TAny](FSharpFunc`2 operation, FSharpFunc`2 handle) in /_//AltCover.Engine/Output.fs:line 23
   at AltCover.CommandLine.doPathOperation@424-1.Invoke(FSharpFunc`2 f, a defaultValue, Boolean store)
   at AltCover.Main.I.doInstrumentation(String[] arguments) in /_//AltCover.Engine/Main.fs:line 717
```

**Comment by SteveGilham** [comment]
This is an interim build that adds some general purpose handling of errors like this one; in particular it adds a catch/rethrow at the `AltCover.Visitor.stateful@1518-1.Invoke(T node)` level and adds the context of the failure to the exception message -

[altcover.8.3.720-github-pre.nupkg.zip](https://github.com/SteveGilham/altcover/files/9709711/altcover.8.3.720-github-pre.nupkg.zip)

In this particular case, as the failing is through `AltCover.OpenCover.visitModule@140-13`, the context will be the assembly that is being examined when the exception was raised, and whether it is being instrumented or may have tracked methods.

**Comment by SteveGilham** [comment]
This extra reporting is now in released build 8.4.840
